full-service/src/service/transaction_builder.rs

```diff
@@ -228,7 +229,11 @@ impl<FPR: FogPubkeyResolver + 'static> WalletTransactionBuilder<FPR> {
         let (fee, fee_token_id) = self.fee.unwrap_or((Mob::MINIMUM_FEE, Mob::ID));
         let fee_amount = Amount::new(fee, fee_token_id);
         let fog_resolver = self.get_fog_resolver(conn)?;
-        let memo_builder = memo.memo_builder(account)?;
+
+        let memo_builder = match account.account_key() {
+            Ok(account_key) => memo.memo_builder(&account_key),
+            Err(_) => Box::<EmptyMemoBuilder>::default(),
```

Comment:
Should probably log this if possible, or maybe just return an error? When is `account.account_key()` expected to fail?

Reply:
When it is a view only account